test/app.e2e-spec.ts

```diff
@@ -15,10 +16,59 @@ describe('AppController (e2e)', () => {
     await app.init();
   });
 
-  it('/ (GET)', () => {
-    return request(app.getHttpServer())
-      .get('/')
-      .expect(200)
-      .expect('Hello World!');
+  const product = {
+    name: 'Coca Cola',
+    weight: 3.1,
+    imageUrl: 'http://image-1.jpg',
+  };
+
+  const productNew = {
+    name: 'Soda',
+    weight: 2.75,
+    imageUrl: 'http://image-2.jpg',
+  };
+
+  let productId;
+
+  it('POST /product', async () => {
+    const res = await request(app.getHttpServer())
+      .post('/products')
+      .send(product)
+      .expect(201);
+
+    expect(res.body.name).toEqual(product.name);
+    expect(res.body.weight).toEqual(product.weight);
+    expect(res.body.imageUrl).toEqual(product.imageUrl);
+
+    productId = res.body._id;
+  });
+
+  it('GET /product/:id', async () => {
+    const res = await request(app.getHttpServer())
+      .get(`/products/${productId}`)
+      .expect(200);
+
+    expect(res.body.name).toEqual(product.name);
+    expect(res.body.weight).toEqual(product.weight);
+    expect(res.body.imageUrl).toEqual(product.imageUrl);
+  });
+
+  it('PUT /product/:id', async () => {
+    const res = await request(app.getHttpServer())
+      .put(`/products/${productId}`)
+      .send(productNew)
+      .expect(200);
+
+    expect(res.body.name).toEqual(productNew.name);
+    expect(res.body.weight).toEqual(productNew.weight);
+    expect(res.body.imageUrl).toEqual(productNew.imageUrl);
   });
+
+  it('DELETE /product/:id', async () => {
+    await request(app.getHttpServer())
+      .delete(`/products/${productId}`)
+      .expect(200);
+  });
+
+  afterAll(() => mongoose.disconnect());
```

Comment:
why do you need this? I guess when the server closes, it also disconnects from mongo, right? It's not the responsibility of the test to do that.

Reply:
This is necessary, as the test is not closing the connection automatically, and because of this the test is unable to pass to the next steps.